### PR TITLE
CPR-79 Fix row updated deleted error on merge

### PR DESCRIPTION
**Pull request by H-Iwanejko**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/RetryExecutor.kt [hunk 1]
```diff
@@ -36,6 +38,7 @@ object RetryExecutor {
         when {
           e::class in exceptions -> {
             lastException = e
+            log.debug("Retrying on error: ${e.message}")
```
Comment (H-Iwanejko) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/RetryExecutor.kt:41]: Won't log in envs as they are set to info. Good for local development to see if the message is consistently retrying on persistence management errors.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/TelemetryService.kt [hunk 2]
```diff
@@ -79,6 +80,11 @@ class TelemetryService(private val telemetryClient: TelemetryClient) {
 
   fun trackEvent(eventType: TelemetryEventType, customDimensions: Map<EventKeys, String?>) {
     val transformedDimensions = customDimensions.entries.associate { it.key.name to it.value }
+    log.debug("Sending telemetry event: ${eventType.name}")
```
Comment (H-Iwanejko) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/TelemetryService.kt:83]: Just added for ease of following events locally, was here previously

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/message/MergeService.kt [hunk 3]
```diff
@@ -72,7 +72,6 @@ class MergeService(
   private fun handleSourceUuidWithSingleRecord(mergeEvent: MergeEvent, sourcePersonEntity: PersonEntity, targetPersonEntity: PersonEntity) {
     mergeRecord(mergeEvent, sourcePersonEntity, targetPersonEntity) { sourcePerson, targetPerson ->
       linkSourceUuidToTargetAndMarkAsMerged(sourcePerson!!, targetPerson)
-      sourcePerson.personKey?.let { personKeyRepository.save(it) }
```
Comment (H-Iwanejko) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/message/MergeService.kt:75]: Not needed as saved in method above (duplication) causing consistent `row updated or deleted`

**Review by markreesmoj — APPROVED** [review]
👍

**Pull request merged**